server/aws-lsp-codewhisperer/src/language-server/inline-completion/codeWhispererServer.ts

```diff
@@ -905,3 +905,27 @@ export const CodewhispererServerFactory =
 
 export const CodeWhispererServerIAM = CodewhispererServerFactory(getOrThrowBaseIAMServiceManager)
 export const CodeWhispererServerToken = CodewhispererServerFactory(getOrThrowBaseTokenServiceManager)
+
+const getLanguageIdFromUri = (uri: string, logging?: any): string => {
+    try {
+        const extension = uri.split('.').pop()?.toLowerCase()
+        return ABAP_EXTENSIONS.has(extension || '') ? 'abap' : ''
+    } catch (err) {
+        logging?.log(`Error parsing URI: ${uri}: ${err}`)
```

Comment:
there are too many uri's that get parsed in this repo, would be helpful to make it more specific here. Maybe `Error parsing URI to determine language`